cub/cub/device/dispatch/tuning/tuning_reduce.cuh

```diff
@@ -279,6 +372,94 @@ struct policy_hub
+  // selection mechanism elsewhere
+
+  [[nodiscard]] _CCCL_API constexpr auto operator()(int arch) const -> reduce_arch_policy
+  {
+    if (arch >= 1000)
+    {
+      // use tuning if we have one, otherwise, pick it from arch 600
+      agent_reduce_policy rp{};
+      if (auto t = get_sm100_tuning(at, ot, os, classify_accum_size(accum_size)))
+      {
+        auto [scaled_items, scaled_threads] = scale_mem_bound(t->threads, t->items, accum_size);
+        rp                                  = agent_reduce_policy{
+          scaled_threads, scaled_items, t->items_per_vec_load, BLOCK_REDUCE_WARP_REDUCTIONS, LOAD_LDG};
+      }
+      else
+      {
+        rp = (*this)(600).reduce_policy;
+      }
+
+      const auto rp_nondet = agent_reduce_policy{
+        rp.block_threads,
+        rp.items_per_thread,
+        rp.vector_load_length,
+        BLOCK_REDUCE_WARP_REDUCTIONS_NONDETERMINISTIC,
+        rp.load_modifier};
+      return {rp, rp, rp, rp_nondet};
+    }
+
+    if (arch >= 600)
+    {
+      constexpr int threads_per_block  = 256;
+      constexpr int items_per_thread   = 16;
+      constexpr int items_per_vec_load = 4;
+
+      // ReducePolicy (P100: 591 GB/s @ 64M 4B items; 583 GB/s @ 256M 1B items)
+      auto [scaled_items, scaled_threads] = scale_mem_bound(threads_per_block, items_per_thread, accum_size);
+      const auto rp =
+        agent_reduce_policy{scaled_threads, scaled_items, items_per_vec_load, BLOCK_REDUCE_WARP_REDUCTIONS, LOAD_LDG};
+
+      const auto rp_nondet = agent_reduce_policy{
+        rp.block_threads,
+        rp.items_per_thread,
+        rp.vector_load_length,
+        BLOCK_REDUCE_WARP_REDUCTIONS_NONDETERMINISTIC,
+        rp.load_modifier};
+      return {rp, rp, rp, rp_nondet};
+    }
+
+    // base policy is for 500
+    // GTX Titan: 255.1 GB/s @ 48M 4B items; 228.7 GB/s @ 192M 1B items
+    constexpr int threads_per_block  = 256;
+    constexpr int items_per_thread   = 20;
+    constexpr int items_per_vec_load = 4;
+
+    auto [scaled_items, scaled_threads] = scale_mem_bound(threads_per_block, items_per_thread, accum_size);
+    const auto rp =
+      agent_reduce_policy{scaled_threads, scaled_items, items_per_vec_load, BLOCK_REDUCE_WARP_REDUCTIONS, LOAD_LDG};
+
+    const auto rp_nondet = agent_reduce_policy{
+      rp.block_threads,
+      rp.items_per_thread,
+      rp.vector_load_length,
+      BLOCK_REDUCE_WARP_REDUCTIONS_NONDETERMINISTIC,
+      rp.load_modifier};
+    return {rp, rp, rp, rp_nondet};
+  }
+};
+
+// stateless version which can be passed to kernels
+template <typename AccumT, typename OffsetT, typename ReductionOpT>
+struct arch_policies_from_types
+{
+  [[nodiscard]] _CCCL_API constexpr auto operator()(int arch) const -> reduce_arch_policy
+  {
+    constexpr auto policies = arch_policies{
+      classify_accum_type<AccumT>(), classify_op<ReductionOpT>(), classify_offset_size<OffsetT>(), int{sizeof(AccumT)}};
```

Comment:
Re: the above comment about potential ABI breaks - this would then just turn into a construction of an environment from 4 "runtime"-valued queries, with the solution I proposed above.